plugins/debuginfo-install.py

```diff
@@ -74,59 +75,126 @@ def configure(self):
+                    if solution["nevra"] is None:
+                        print("TODO We have this package and it may be the wrong one: %s", package)
+
+                    if not self._install_debug(package.debug_name, solution["nevra"]):
+                        self.debuginfo_missing.add(str(package)) # TODO check this is the correct package being logged - could be different architecture etc.
+
+                    if not self._install_debug(package.debugsource_name, solution["nevra"]):
+                        self.debugsource_missing.add(str(package))
+
+        if self.debuginfo_missing:
+            logger.info(_("Could not find debuginfo packages for the following packages: %s") %
+                        ", ".join(sorted(self.debuginfo_missing)))
+
+        if self.debugsource_missing:
+            logger.info(_("Could not find debugsource packages for the following packages: %s") %
+                        ", ".join(sorted(self.debugsource_missing)))
 
         if errors_spec and self.base.conf.strict:
             raise dnf.exceptions.PackagesNotAvailableError(_("Unable to find a match"),
                                                            pkg_spec=' '.join(errors_spec))
 
-    def _dbg_available(self, dbgname, package, match_evra):
-        if match_evra:
-            return self.packages_available.filter(
-                name=dbgname,
-                epoch=int(package.epoch),
-                version=str(package.version),
-                release=str(package.release),
-                arch=str(package.arch))
-        else:
-            return self.packages_available.filter(
-                name=dbgname,
-                arch=str(package.arch))
-
-    def _di_install(self, package):
-        for dbgname in [package.debug_name, package.source_debug_name]:
-            if dbgname in self.dbgdone:
-                break
-            if self._dbg_available(dbgname, package, True):
-                di = "{0}-{1}:{2}-{3}.{4}".format(
-                    dbgname,
-                    package.epoch,
-                    package.version,
-                    package.release,
-                    package.arch)
-                self.base.install(di)
-            elif self._dbg_available(dbgname, package, False):
-                di = "{0}.{1}".format(dbgname, package.arch)
-                self.base.install(di)
-            else:
-                continue
-            self.dbgdone.append(dbgname)
-            break
-        else:
-            logger.info(
-                _("Could not find debuginfo for package: %s") % package)
+    def _install_debug_from_system(self, package, debug_name):
+        query = self.packages_available.filter(name=debug_name,
+                                               epoch=package.epoch,
+                                               version=package.version,
+                                               release=package.release,
+                                               arch=package.arch)
```

Comment:
Please replace self.packages_available by self.base.sack.query() to prevent reinstalling of packages